src/objects/triangle.cpp

```diff
@@ -0,0 +1,47 @@
+//
+// Created by andrew on 28/04/23.
+//
+#include "objects/triangle.hpp"
+
+Triangle::Triangle(const Vertex &v1, const Vertex &v2, const Vertex &v3) {
+    vertices_m[0] = v1;
+    vertices_m[1] = v2;
+    vertices_m[2] = v3;
+
+    e1_m = vertices_m[1].position - vertices_m[0].position;
+    e2_m = vertices_m[2].position - vertices_m[0].position;
+    normal_m = glm::normalize(glm::cross(e1_m, e2_m));
+}
+
+bool Triangle::testIntersections(const Ray &cast_ray, glm::vec3 &int_point, glm::vec3 &loc_normal) const {
+    constexpr double kEpsilon = 0.0000001;
```

Comment:
Guess you can keep epsilon consistent across all project, no need to have such precision
```suggestion
    constexpr double kEpsilon = 0.0001;
```

Reply:
Updated in 1e821ca